source/index.ts

```diff
@@ -4,49 +4,73 @@
 
 export type Class<T = unknown> = new (...args: any[]) => T;
 
-export const enum TypeName {
-	null = 'null',
-	boolean = 'boolean',
-	undefined = 'undefined',
-	string = 'string',
-	number = 'number',
-	bigint = 'bigint',
-	symbol = 'symbol',
-	Function = 'Function',
-	Generator = 'Generator',
-	AsyncGenerator = 'AsyncGenerator',
-	GeneratorFunction = 'GeneratorFunction',
-	AsyncGeneratorFunction = 'AsyncGeneratorFunction',
-	AsyncFunction = 'AsyncFunction',
-	Observable = 'Observable',
-	Array = 'Array',
-	Buffer = 'Buffer',
-	Object = 'Object',
-	RegExp = 'RegExp',
-	Date = 'Date',
-	Error = 'Error',
-	Map = 'Map',
-	Set = 'Set',
-	WeakMap = 'WeakMap',
-	WeakSet = 'WeakSet',
-	Int8Array = 'Int8Array',
-	Uint8Array = 'Uint8Array',
-	Uint8ClampedArray = 'Uint8ClampedArray',
-	Int16Array = 'Int16Array',
-	Uint16Array = 'Uint16Array',
-	Int32Array = 'Int32Array',
-	Uint32Array = 'Uint32Array',
-	Float32Array = 'Float32Array',
-	Float64Array = 'Float64Array',
-	BigInt64Array = 'BigInt64Array',
-	BigUint64Array = 'BigUint64Array',
-	ArrayBuffer = 'ArrayBuffer',
-	SharedArrayBuffer = 'SharedArrayBuffer',
-	DataView = 'DataView',
-	Promise = 'Promise',
-	URL = 'URL'
+const objectTypeNames = [
+	'Function',
+	'Generator',
+	'AsyncGenerator',
+	'GeneratorFunction',
+	'AsyncGeneratorFunction',
+	'AsyncFunction',
+	'Observable',
+	'Array',
+	'Buffer',
+	'Object',
+	'RegExp',
+	'Date',
+	'Error',
+	'Map',
+	'Set',
+	'WeakMap',
+	'WeakSet',
+	'ArrayBuffer',
+	'SharedArrayBuffer',
+	'DataView',
+	'Promise',
+	'URL'
+] as const;
+
+type ObjectTypeName = typeof objectTypeNames[number];
+
+const primitives = {
+	null: null,
+	undefined,
+	string: String(),
+	number: Number(),
+	bigint: BigInt(1),
+	boolean: Boolean(),
+	symbol: Symbol('')
+} as const;
+
+type PrimitiveTypeName = keyof typeof primitives;
+export type Primitive = typeof primitives[PrimitiveTypeName];
+
+function isPrimitiveTypeName(v: any): v is PrimitiveTypeName {
+	return Object.prototype.hasOwnProperty.call(primitives, v);
 }
 
+const typedArrays = {
+	Int8Array: new Int8Array(),
+	Uint8Array: new Uint8Array(),
+	Uint8ClampedArray: new Uint8ClampedArray(),
+	Int16Array: new Int16Array(),
+	Uint16Array: new Uint16Array(),
+	Int32Array: new Int32Array(),
+	Uint32Array: new Uint32Array(),
+	Float32Array: new Float32Array(),
+	Float64Array: new Float64Array(),
+	BigInt64Array: new BigInt64Array(),
+	BigUint64Array: new BigUint64Array()
```

Comment:
This comes with a slight overhead. I prefer:

```ts
export type TypedArray =
	| Int8Array
	| Uint8Array
	| Uint8ClampedArray
	| Int16Array
	| Uint16Array
	| Int32Array
	| Uint32Array
	| Float32Array
	| Float64Array
	| BigInt64Array
	| BigUint64Array;
```

Reply:
The performance impact should be negligible, but if you'd prefer two separate lists that you'd like to maintain I can do that.

Reply:
Yes, I prefer that. It's not like it will change very often.